xilem_core/src/view.rs

```diff
@@ -69,6 +64,9 @@ pub trait View<State, Action, Context: ViewPathTracker>: ViewMarker + 'static {
     /// This often means routing information for messages to child views or view sequences,
     /// to avoid sending outdated views.
     /// This is also used in [`memoize`](crate::memoize) to store the previously constructed view.
+    ///
+    /// The type used for this associated type cannot be treated as public API; this is
+    /// internal state to the `ViewSequence` implementation.
```

Comment:
Can it not? I mean that's probably the average case, that's true. But I don't see why it *cannot* be treated as public API (in rare cases).

Also:
```suggestion
    /// internal state to the `View` implementation.
```

Reply:
What I mean by this is that implementations are allowed to change their state type in minor releases.

Like, obviously we can't stop people using it as public API. But they shouldn't make a habit of it.

Reply:
Ok, makes sense.

Maybe worth mentioning this with an example to enforce this (like the `mod view { pub struct MyViewState; }` "trick")